src/mono/mono/component/marshal-ilgen-stub.c

```diff
@@ -14,6 +14,37 @@ stub_emit_marshal_ilgen (EmitMarshalContext *m, int argnum, MonoType *t,
 		MonoMarshalSpec *spec, int conv_arg,	
 		MonoType **conv_arg_type, MarshalAction action, MonoMarshalLightweightCallbacks* lightweight_cb)
 {
+	if (spec && spec->native == MONO_NATIVE_CUSTOM)
+		return conv_arg;
+
+	if (spec && spec->native == MONO_NATIVE_ASANY)
+		return conv_arg;
+
+	switch (t->type) {
+	case MONO_TYPE_BOOLEAN:
+		return lightweight_cb->emit_marshal_scalar (m, argnum, t, spec, conv_arg, conv_arg_type, action);
```

Comment:
I'm not sure we need to do `char` and `bool` here...  looking at `mono_emit_marshal` we only get called if `m->runtime_marshalling_enabled` which means we don't have `DisableRuntimeMarshallingAttribute` (https://github.com/dotnet/runtime/issues/60639) on the assembly, so `bool` and `char` are non-blittable.